### PR TITLE
Port to Smalltalk/X - part 1

### PR DESCRIPTION
**Pull request by janvrany**
This is a first portion of changes require to load Powerlang into Smalltalk/X. 

This PR contains changes to various methods here and there in order to make
code more portable and working on Smalltalk/X. It does to change any functionality. 

FYI: @melkyades

### CODE REVIEW

File: bootstrap/src/Powerlang-Core/SSmalltalkScanner.class.st [hunk 1]
```diff
@@ -135,10 +135,12 @@ SSmalltalkScanner >> nextColon [
 SSmalltalkScanner >> nextComment [
 	| start comment |
 	start := stream position.
-	(stream skipTo: $") ifFalse: [self error: 'unfinished comment' at: start].
+	[ stream atEnd not and:[stream peek ~~ $"] ] whileTrue: [stream next].
```
Comment (melkyades) [inline bootstrap/src/Powerlang-Core/SSmalltalkScanner.class.st:139]: is ~~ the same than !== ? in that case please use !==, we don't use ~= or ~~ in this code base

File: bootstrap/src/Powerlang-Tests/KernelBuilderTest.class.st [hunk 2]
```diff
@@ -99,8 +99,7 @@ KernelBuilderTest >> test220writeSegmentWithBase [
 			equals: b + offset + 16r10;
 		assert: (writer encodedReferenceTo: falseObj)
 			equals: b + offset + 16r18.
-	FileLocator temp asFileReference / 'kernel.ims'
-		binaryWriteStreamDo: [ :s | s nextPutAll: writer image ]
+	ByteArray streamContents: [ :s | s nextPutAll: writer image ]
```
Comment (melkyades) [inline bootstrap/src/Powerlang-Tests/KernelBuilderTest.class.st:102]: If we are not writing to a file, what's the point of writing at all?

File: bootstrap/src/Powerlang-Core/OptimizingInstructionFragmenter.class.st [hunk 3]
```diff
@@ -244,7 +244,7 @@ OptimizingInstructionFragmenter >> fragmentTernary: instruction [
 	left addUse: new at: #left.
 	right addUse: new at: #right.
 	self
-		ASSERT: (#(_basicAt:put: _basicULongAt:put:	_byteAt:put: _objectAtOffset:put:)
+		ASSERT: (#(#'_basicAt:put:' #'_basicULongAt:put:'     #'_byteAt:put:' #'_objectAtOffset:put:')
```
Comment (melkyades) [inline bootstrap/src/Powerlang-Core/OptimizingInstructionFragmenter.class.st:247]: doesn't this work on stx? because it just adds noise

**Comment by melkyades** [comment]
Minor comments to check before merging

File: bootstrap/src/Powerlang-Core/SSmalltalkScanner.class.st (hunk 1 above)
Comment (janvrany) [reply bootstrap/src/Powerlang-Core/SSmalltalkScanner.class.st]: Good catch! Fixed in fa0fbbf.

File: bootstrap/src/Powerlang-Core/OptimizingInstructionFragmenter.class.st (hunk 3 above)
Comment (janvrany) [reply bootstrap/src/Powerlang-Core/OptimizingInstructionFragmenter.class.st]: It did not but then I found this on so many places I fixed the parser and then forgot to revert this change. 
Reverted in df7267a

File: bootstrap/src/Powerlang-Tests/KernelBuilderTest.class.st (hunk 2 above)
Comment (janvrany) [reply bootstrap/src/Powerlang-Tests/KernelBuilderTest.class.st:102]: Perhaps making sure writer does not crash? 
What was the point of writing it to file and then never check (and remove!) the file?

**Pull request merged**